### PR TITLE
[ Add Macropad 15Pad ]

### PR DESCRIPTION
**Pull request by veriyaniot**
## Description
<p align="center">
  
</p>

**[ Add Macropad 15Pad ]** 

**Thanks to [zlabkeeb](https://github.com/zlabkeeb)  for trusting me to work on this project**

- RP2040 Zero
- Hotswap Kailh
- 12 Switch
- 3 Rotary Knob

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/zlabkeeb/15pad/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * ├─────────┼─────────┼─────────┼─────────┤
+     * │    1    │    2    │    3    │    4    │
+     * ├─────────┼─────────┼─────────┼─────────┤
+     * │    5    │    6    │    7    │    8    │
+     * ├─────────┼─────────┬─────────┼─────────┤
+     * │    9    │    0    │ ENTER   │  DELL   │
+     * ├─────────┼─────────┬─────────┼─────────┤
+     */
+    [0] = LAYOUT(
+        KC_MUTE, TO(0),    TO(1),
```
Comment (drashna) [inline keyboards/zlabkeeb/15pad/keymaps/default/keymap.c:33]: Just a nitpick here, but most of the keycodes have a 7 character version, so having it spaced as such would keep consistent spacing and alignment.

It's not mandatory but does look better.

File: keyboards/zlabkeeb/15pad/rules.mk
Comment (drashna) [inline keyboards/zlabkeeb/15pad/rules.mk:1]: This can be removed

Comment (veriyaniot) [reply keyboards/zlabkeeb/15pad/rules.mk:1]: if it is removed, will it be okay from building failures?

File: keyboards/zlabkeeb/15pad/keymaps/default/keymap.c (hunk 1 above)
Comment (veriyaniot) [reply keyboards/zlabkeeb/15pad/keymaps/default/keymap.c:33]: what should i do to look better??

is this enough?

File: keyboards/zlabkeeb/15pad/rules.mk
Comment (veriyaniot) [reply keyboards/zlabkeeb/15pad/rules.mk:1]: [Build Failures](https://github.com/qmk/qmk_firmware/pull/23257/checks?check_run_id=22692108259)

File: keyboards/zlabkeeb/15pad/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+        )
+};
+#if defined (ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_LEFT, KC_RIGHT), ENCODER_CCW_CW(KC_DOWN, KC_UP)},
+    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS)}
```
Comment (waffle87) [inline keyboards/zlabkeeb/15pad/keymaps/default/keymap.c:48]:
```suggestion
        KC_MUTE, KC_MPLY, KC_TRNS,
        KC_1, KC_2, KC_3, KC_4,
        KC_5, KC_6, KC_7, KC_8,
        KC_9, KC_0, KC_ENT, KC_DEL
        )
};
#if defined (ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_LEFT, KC_RIGHT), ENCODER_CCW_CW(KC_DOWN, KC_UP)}
```

File: keyboards/zlabkeeb/15pad/readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# 15PAD
+
+![15PAD](https://i.imgur.com/J7sZSnxh.jpeg)
```
Comment (waffle87) [inline keyboards/zlabkeeb/15pad/readme.md:3]: This requirement has been changed, as it seems no longer supported by Imgur.
```suggestion
![15PAD](https://i.imgur.com/J7sZSnx.jpeg)
```

Comment (veriyaniot) [reply keyboards/zlabkeeb/15pad/readme.md]: okay thanks

File: keyboards/zlabkeeb/15pad/keymaps/default/keymap.c (hunk 2 above)
Comment (veriyaniot) [reply keyboards/zlabkeeb/15pad/keymaps/default/keymap.c]: thanks!

**Review by waffle87 — APPROVED** [review]

File: keyboards/zlabkeeb/15pad/readme.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    make zlabkeeb/15pad:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 1 way:
+
+-   **Physical reset button**: Briefly press the button on the top of the PCB
```
Comment (zvecr) [inline keyboards/zlabkeeb/15pad/readme.md:25]:
```suggestion
Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the top of the PCB
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available

```

**Comment by veriyaniot** [comment]
what else needs to be improved??
I'm very enthusiastic 😅

**Review by drashna — APPROVED** [review]

**Comment by veriyaniot** [comment]
Thanks @drashna

**Pull request merged**